lib/finch/request.ex

```diff
@@ -49,7 +49,17 @@ defmodule Finch.Request do
   """
   @type body() :: iodata() | {:stream, Enumerable.t()} | nil
 
-  @type t :: %Finch.Request{}
+  @type t :: %__MODULE__{
+          scheme: Mint.Types.scheme(),
+          host: String.t(),
+          port: :inet.port_number(),
+          method: String.t(),
+          path: String.t(),
+          headers: headers(),
+          body: body(),
+          query: String.t(),
```

Comment:
```suggestion
          query: String.t() | nil,
```

Reply:
Actually the "nullable" fields are:
- host (from `URI.parse()`)
- port (from `URI.parse()`)
- body (from API contract)
- query (from `URI.parse()`)
- unix_socket (from API contract)

Maybe `Finch.Request.parse_url/1` should validate also `host` (to be not `nil`).

Reply:
Can you provide an example `Finch.build/5` call that will result in `nil` for host? It does not appear to be possible since we require a scheme to be present.

Reply:
```!elixir
Finch.Request.build(:get, "http:", [], nil, [])          
%Finch.Request{
  body: nil,
  headers: [],
  host: nil,
  method: "GET",
  path: "/",
  port: 80,
  query: nil,
  scheme: :http,
  unix_socket: nil
}
```

Reply:
Ah of course! I had not tried without `//`. In this case, I suppose it would make sense to raise an error if the host were parsed as `nil`, just like we raise when the scheme is invalid/unsupported.

Reply:
I believe the host is ignored when using the unix domain socket so maybe it is fine to leave this as is, and, if anything, let Mint raise if appropriate?

Reply:
Good point, thanks @wojtekmach ! Let's leave it as is.